### PR TITLE
Utopia-1264 Modals not updated moving from Final Review > MPO review …

### PR DESCRIPTION
**Pull request by antsand**
…> Edit in prog/Incomplete

# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/utils/status.ts [hunk 1]
```diff
@@ -73,275 +91,310 @@ const defaultFinalReviewModal: Modal = {
...
+      banner: BannerText.InCompleteStatusCalloutText.Drafter.en,
+      modal: defaultIncompleteModal,
+      Privileges: {
+        MPO: {
+          banner: BannerText.InCompleteStatusCalloutText.Drafter.en, //incomplete only have one banner for both mpo and drafter
+          changeStatus: [
+            {
+              status: 'EDIT_IN_PROGRESS',
+              modal: checkReviewStatusDelegate(pia)
+                ? resetReviewEditInPRogressModal
```
Comment (kushal-arora-fw) [inline src/frontend/src/utils/status.ts:185]: changing from INCOMPLETE to EDIT_IN_PROGRESS shouldn't require the unlock modal.

File: src/frontend/src/utils/status.ts [hunk 2]
```diff
@@ -73,275 +91,310 @@ const defaultFinalReviewModal: Modal = {
...
+                : defaultEditInProgressModal,
+            },
+            {
+              status: 'CPO_REVIEW',
+              modal: defaultCPOReviewModal,
+            },
```
Comment (kushal-arora-fw) [inline src/frontend/src/utils/status.ts:191]: is changing to CPO_REVIEW from INCOMPLETE even allowed?

File: src/frontend/src/utils/status.ts [hunk 3]
```diff
@@ -73,275 +91,310 @@ const defaultFinalReviewModal: Modal = {
...
+            {
+              status: 'EDIT_IN_PROGRESS',
+              modal: defaultEditInProgressModal,
+            },
+            {
+              status: 'CPO_REVIEW',
+              modal: defaultCPOReviewModal,
+            },
+          ],
+        },
```
Comment (kushal-arora-fw) [inline src/frontend/src/utils/status.ts:209]: in an incomplete status, no extra privileges are allowed.

File: src/frontend/src/utils/status.ts (hunk 2 above)
Comment (antsand) [reply src/frontend/src/utils/status.ts:191]: I think CPO can do it. I haven't changed any of this logic.

File: src/frontend/src/utils/status.ts (hunk 1 above)
Comment (antsand) [reply src/frontend/src/utils/status.ts:185]: correct. The check should fail as the review should be empty. Having the check should not cause any harm unless the logic is broken somewhere else.

File: src/frontend/src/utils/status.ts (hunk 3 above)
Comment (antsand) [reply src/frontend/src/utils/status.ts:209]: I believe CPO does have the privilege to change things. I haven't changed this logic as such.

File: src/frontend/src/pages/PIAForm/index.tsx [hunk 4]
```diff
@@ -277,6 +278,16 @@ const PIAFormPage = () => {
...
   // Event Handlers
   //
+
+  const populateModalFn = (modal: object, status: string) => {
+    setPiaModalTitleText(Object(modal).title);
+    setPiaModalParagraph(Object(modal).description);
+    setPiaModalConfirmLabel(Object(modal).confirmLabel);
+    setPiaModalCancelLabel(Object(modal).cancelLabel);
+    //setStatusLocal(status);
+    //setShowModal(true);
```
Comment (binggaofw) [inline src/frontend/src/pages/PIAForm/index.tsx:288]: commets can be deleted

**Review by binggaofw — APPROVED** [review]
looks good to me, approve

**Pull request merged**